ProjectDplusDsSparse.py

```diff
@@ -163,16 +174,28 @@
             hVarPrompt.SetName(f'hPrompt{varName}_{ptLowLabel:.0f}_{ptHighLabel:.0f}')
             prompt_dict[iVar].append(hVarPrompt)
             hVarPrompt.Write()
-            hVarFD = sparseReco['RecoFD'].Projection(axisNum)
             # apply pt weights
-            if iVar == 'Pt' and (args.ptweightsB or args.ptweights):
-                for iBin in range(1, hVarFD.GetNbinsX()+1):
-                    if hVarFD.GetBinContent(iBin) > 0.:
-                        relStatUnc = hVarFD.GetBinError(iBin) / hVarFD.GetBinContent(iBin)
-                        ptCent = hVarFD.GetBinCenter(iBin)
-                        hVarFD.SetBinContent(iBin, hVarFD.GetBinContent(iBin) * sPtWeightsRecoDfromB(ptCent))
-                        hVarFD.SetBinError(iBin, hVarFD.GetBinContent(iBin) * relStatUnc)
-            hVarFD.SetName(f'hFD{varName}_{ptLowLabel:.0f}_{ptHighLabel:.0f}')
+            if args.ptweightsB and args.exactptweightsB:
```

Comment:
Here you are applying the pT weights also for `iVar = 'InvMass'`

Reply:
That was intended, to have a consistent reweighting (it could have an effect in the comparison of the width between data and MC), but I now realised that we do not apply it for the prompt, so no strong opinon

Reply:
Not sure if the pT(B) weights could create visible effects on the invariant-mass width. In any case, we should be consistent between prompt and FD

Reply:
In principle, if you have a softer pT distribution and an increasing sigma vs pT you should get lower sigma values if the pT is reweighted (both for prompt and FD), but the effect should be small, so ok for me to be consistent